File: loop_benchmark.php
<?php

use parallel\{Runtime, Channel};

$totalLoops = 0;
$runtimes = [];
$futures = [];
$cpuCount = shell_exec("nproc");
$channel = Channel::make('loopCounter', Channel::Infinite);

$sigint = function () use (&$totalLoops, &$runtimes, &$futures, $channel) {
    echo "PHP " . PHP_VERSION . " looped " . number_format($totalLoops) . " times.\n";
    foreach ($futures as $i => $future) {
        $future->cancel();
        $runtimes[$i]->close();
    }
    $channel->close();
    exit(0);
};

pcntl_async_signals(true);
pcntl_signal(SIGINT, $sigint);

$run = function ($channel) {
    pcntl_async_signals(true);
    pcntl_signal(SIGINT, function () {
    });

    $sendRuntime = new Runtime();
    $send = function ($value) use ($channel) {
        $channel->send($value);
    };

    while (true) {
        for ($j = 0; $j < 5_000_000; ++$j) {
            // TODO: CPU busy work here
        }
        $sendRuntime->run($send, [$j]);
    }
};

for ($i = 0; $i < $cpuCount / 2; ++$i) {
    $runtimes[$i] = new Runtime();
    $futures[$i] = $runtimes[$i]->run($run, [$channel]);
}

while ($totalLoops += $channel->recv());

File: loop_benchmark.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using System.Diagnostics;
using System.Globalization;

class Program
{
    private static long totalLoops = 0;
    private static CancellationTokenSource cts = new CancellationTokenSource();

    static async Task Main(string[] args)
    {
        int cpuCount = Environment.ProcessorCount;

        Console.CancelKeyPress += (sender, eventArgs) =>
        {
            Console.WriteLine($"C# .NET looped {totalLoops.ToString("N0", CultureInfo.InvariantCulture)} times.");
            cts.Cancel();
            eventArgs.Cancel = true;
        };

        Task[] tasks = new Task[cpuCount];

        for (int i = 0; i < cpuCount; i++)
        {
            tasks[i] = Task.Run(() => DoWork(cts.Token));
        }

        try
        {
            await Task.WhenAll(tasks);
        }
        catch (OperationCanceledException)
        {
        }
    }

    static void DoWork(CancellationToken token)
    {
        while (!token.IsCancellationRequested)
        {
            // Simulate CPU work
            for (int j = 0; j < 5_000_000; j++)
            {
                // TODO: busy work here
            }
            Interlocked.Add(ref totalLoops, 5_000_000);
        }
    }
}
